tests/other/test_smtlibv2.py

```diff
@@ -26,82 +27,362 @@
+        self.assertEqual(c[4], ord("E"))
+        self.assertRaises(IndexError, c.get, 5)
+
+    def test_ConstantArrayBitvec2(self):
+        c = MutableArray(ArrayVariable(index_size=32, value_size=8, length=5, name="ARR"))
+        c[1] = 10
+        c[2] = 20
+        c[3] = 30
+        self.assertEqual(c[1], 10)
+        self.assertEqual(c[2], 20)
+        self.assertEqual(c[3], 30)
+        c[2] = 100
+        self.assertEqual(c[2], 100)
+
+    def test_ArrayDefault3(self):
+        c = MutableArray(
+            ArrayVariable(index_size=32, value_size=8, length=5, default=0, name="ARR")
         )
-        self.assertEqual(translate_to_smtlib(constraint), translate_to_smtlib(new_constraint))
+        self.assertEqual(c[1], 0)
+        self.assertEqual(c[2], 0)
+        self.assertEqual(c[3], 0)
 
-        Z3Solver.instance().can_be_true.cache_clear()
-        self.assertEqual(ground_truth, Z3Solver.instance().can_be_true(constraints, new_constraint))
+        c[1] = 10
+        c[3] = 30
+        self.assertEqual(c[1], 10)
+        self.assertEqual(c[2], 0)
+        self.assertEqual(c[3], 30)
 
-        Z3Solver.instance().can_be_true.cache_clear()
-        self.assertEqual(
-            ground_truth,
-            Z3Solver.instance().can_be_true(constraints.related_to(new_constraint), new_constraint),
+    def test_ArrayDefault4(self):
+        cs = ConstraintSet()
+        a = MutableArray(cs.new_array(index_size=32, value_size=8, length=4, default=0, name="ARR"))
+        i = cs.new_bitvec(size=a.index_size)
+        SelectedSolver.instance().must_be_true(cs, 0 == a.default)
+        SelectedSolver.instance().must_be_true(cs, a[i] == a.default)
+        cs.add(i == 2)
+        SelectedSolver.instance().must_be_true(cs, 0 == a.default)
+        SelectedSolver.instance().must_be_true(cs, a[i] == a.default)
+
+        b = a[:]
+        i = cs.new_bitvec(size=a.index_size)
+        SelectedSolver.instance().must_be_true(cs, 0 == b.default)
+        SelectedSolver.instance().must_be_true(cs, b[i] == b.default)
+
+        a[1] = 10
+        a[2] = 20
+        a[3] = 30
+        # a := 0 10 20 30 0 0 x x x x      (x undefined)
+        SelectedSolver.instance().must_be_true(cs, a.default == 0)
+        SelectedSolver.instance().must_be_true(cs, a[0] == 0)
+        SelectedSolver.instance().must_be_true(cs, a[1] == 10)
+        SelectedSolver.instance().must_be_true(cs, a[2] == 20)
+        SelectedSolver.instance().must_be_true(cs, a[3] == 30)
+        # SelectedSolver.instance().must_be_true(cs, a[4] == 0) #undefined!
+
+        b = a[:]
+        # b := 0 10 20 30 0 0 x x x x      (x undefined)
+        SelectedSolver.instance().must_be_true(cs, b.default == 0)
+        SelectedSolver.instance().must_be_true(cs, b[0] == 0)
+        SelectedSolver.instance().must_be_true(cs, b[1] == 10)
+        SelectedSolver.instance().must_be_true(cs, b[2] == 20)
+        SelectedSolver.instance().must_be_true(cs, b[3] == 30)
```

Comment:
I think I'm confused about what's actually being tested here.

`SelectedSolver.instance().must_be_true(cs, expr)` returns a boolean telling us whether the expression _must be true_, right? We don't ever seem to check the return value of these expressions. 

Wouldn't doing `self.assertTrue(SelectedSolver.instance().must_be_true(cs, expr))` be what we actually want?

Am I missing something?